src/Build.UnitTests/BackEnd/RedirectConsoleWriter_Tests.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Execution;
+using Xunit;
+
+namespace Microsoft.Build.Engine.UnitTests.BackEnd
+{
+    public class RedirectConsoleWriter_Tests
+    {
+        [Fact]
+        public async Task EmitConsoleMessages()
+        {
+            StringBuilder sb = new StringBuilder();
+            var writer = OutOfProcServerNode.RedirectConsoleWriter.Create(text => sb.Append(text));
+
+            writer.WriteLine("Line 1");
+            await Task.Delay(300);
```

Comment:
consider setting the awaitable to false

Reply:
Unit tests do not have synchronization context.